packages/purpose-process/src/services/validators.ts

```diff
@@ -163,3 +182,67 @@ export const isArchivable = (purposeVersion: PurposeVersion): boolean =>
 export const isSuspendable = (purposeVersion: PurposeVersion): boolean =>
   purposeVersion.state === purposeVersionState.active ||
   purposeVersion.state === purposeVersionState.suspended;
+
+export async function isLoadAllowed(
+  eservice: EService,
+  purpose: Purpose,
+  purposeVersion: PurposeVersion,
+  readModelService: ReadModelService
+): Promise<boolean> {
+  const consumerPurposes = await readModelService.getAllPurposes({
+    eservicesIds: [eservice.id],
+    consumersIds: [purpose.consumerId],
+    states: [purposeVersionState.active],
+    producersIds: [],
+    excludeDraft: true,
+  });
+
+  const allPurposes = await readModelService.getAllPurposes({
+    eservicesIds: [eservice.id],
+    consumersIds: [],
+    producersIds: [],
+    states: [purposeVersionState.active],
+    excludeDraft: true,
+  });
```

Comment:
In terms of performance, what about doing only one query to get all the purposes and filter the consumer's ones after that?

Reply:
This is a good idea. I ported it 1:1 from Scala, I would first ask @galales or @beetlecrunch if it worth it doing so

Reply:
Updated!